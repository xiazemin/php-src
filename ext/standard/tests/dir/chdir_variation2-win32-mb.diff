011+ string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"
011- string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"
015+ string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"
015- string(%d) "%s私はガラスを食べられますlevel_one"
020+ string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"
020- string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"
029+ string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"
029- string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"
034+ string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"
034- string(%d) "%s私はガラスを食べられますlevel_one"