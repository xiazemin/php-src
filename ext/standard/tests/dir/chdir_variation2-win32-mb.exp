*** Testing chdir() : usage variations ***

-- $directory = './私はガラスを食べられますlevel_one': --
bool(true)
bool(true)
string(%d) "%slevel_one"

-- $directory = '私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two': --
bool(true)
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"

-- $directory = '..': --
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one"

-- $directory = '私はガラスを食べられますlevel_two', '.': --
bool(true)
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"

-- $directory = '../': --
bool(true)
string(%d) "%slevel_one"

-- $directory = './': --
bool(true)
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"

-- $directory = '../../'私はガラスを食べられますlevel_one': --
bool(true)
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one"
===DONE===