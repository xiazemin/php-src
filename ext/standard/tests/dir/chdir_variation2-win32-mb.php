<?php
/* Prototype  : bool chdir(string $directory)
 * Description: Change the current directory 
 * Source code: ext/standard/dir.c
 */

/*
 * Test chdir() with variations of relative paths
 */

echo "*** Testing chdir() : usage variations ***\n";

$base_dir_path = dirname(__FILE__);

$level_one_dir_name = "私はガラスを食べられますlevel_one";
$level_one_dir_path = "$base_dir_path/$level_one_dir_name";

$level_two_dir_name = "私はガラスを食べられますlevel_two";
$level_two_dir_path = "$base_dir_path/$level_one_dir_name/$level_two_dir_name";

// create directories
mkdir($level_one_dir_path);
mkdir($level_two_dir_path);

echo "\n-- \$directory = './私はガラスを食べられますlevel_one': --\n";
var_dump(chdir($base_dir_path));
var_dump(chdir("./$level_one_dir_name"));
var_dump(getcwd());

echo "\n-- \$directory = '私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two': --\n";
var_dump(chdir($base_dir_path));
var_dump(chdir("$level_one_dir_name/$level_two_dir_name"));
var_dump(getcwd());

echo "\n-- \$directory = '..': --\n";
var_dump(chdir('..'));
var_dump(getcwd());

echo "\n-- \$directory = '私はガラスを食べられますlevel_two', '.': --\n";
var_dump(chdir($level_two_dir_path));
var_dump(chdir('.'));
var_dump(getcwd());

echo "\n-- \$directory = '../': --\n";
var_dump(chdir('../'));
var_dump(getcwd());

echo "\n-- \$directory = './': --\n";
var_dump(chdir($level_two_dir_path));
var_dump(chdir('./'));
var_dump(getcwd());

echo "\n-- \$directory = '../../'私はガラスを食べられますlevel_one': --\n";
var_dump(chdir($level_two_dir_path));
var_dump(chdir("../../$level_one_dir_name"));
var_dump(getcwd());

$file_path = dirname(__FILE__);
chdir($file_path);/* not that PWD is accidentialy one of the dirs to be deleted. */
rmdir("$file_path/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two");
rmdir("$file_path/私はガラスを食べられますlevel_one");
?>
===DONE===
