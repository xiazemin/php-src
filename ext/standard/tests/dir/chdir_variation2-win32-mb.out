*** Testing chdir() : usage variations ***

-- $directory = './私はガラスを食べられますlevel_one': --
bool(true)
bool(true)
string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"

-- $directory = '私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two': --
bool(true)
bool(true)
string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"

-- $directory = '..': --
bool(true)
string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"

-- $directory = '私はガラスを食べられますlevel_two', '.': --
bool(true)
bool(true)
string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"

-- $directory = '../': --
bool(true)
string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"

-- $directory = './': --
bool(true)
bool(true)
string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"

-- $directory = '../../'私はガラスを食べられますlevel_one': --
bool(true)
bool(true)
string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"
===DONE===